src/main/java/io/opentracing/contrib/global/concurrent/TracedCallable.java

```
@@ -0,0 +1,124 @@
+    }
+
+    public TracedCallable<T> withOperationName(String operationName) {
+        this.operationName = operationName;
+        return this;
+    }
+
+    public TracedCallable<T> asChildOf(SpanContext parent) {
+        this.parentContext = parent;
+        return this;
+    }
+
+    protected Span startNewSpan() {
+        SpanBuilder spanBuilder = GlobalTracer.tracer().buildSpan(operationName);
+        if (parentContext != null) spanBuilder = spanBuilder.asChildOf(parentContext);
+        return spanBuilder.start();
+    }
+
+    /**
+     * This method {@linkplain #startNewSpan() starts a new span} and calls the delegate, returning the result.
+     * Finally, the started span is {@linkplain Span#close() closed} again.
+     *
+     * @return The result from the delegate call.
+     * @throws Exception in case the delegate call threw an exception
+     *                   or there were exceptions starting or closing a new span.
+     */
+    public T call() throws Exception {
+        // The javadoc doesn't mention the details of the errorhandling
+        // because they are in line with normal try-with-resources logic:
+        // - try to close resources created (started span) whether the call succeeded or failed.
+        // - in case closing fails: a) throw the call exception if there was one (but addSuppressed the close failure)
+        //                       or b) if there was no call exception throw the close failure instead.
+
+        Span newSpan = null;            // so it can be closed
+        Exception callException = null; // so it can be thrown from finally if necessary
+        try {
+
+            newSpan = startNewSpan();
+            return delegate.call();
+
+        } catch (Exception callEx) {
+            callException = callEx;
+            throw callException;
+        } finally {
+            if (newSpan != null) try {
+                newSpan.close();
+            } catch (Exception closeFailure) {
```

this is a lot of defense around the concern of what we do with a possible runtime exception returned from `close()`. I still think this is overengineering. Most libraries don't bother with this, either log it or don't.

---

Should `span.close` throw exception or not. I think that should decide by the tracer implementation. As a globalTracer, we should only control the tracer-instance, spans, and in-process(in-thread/inter-thread) propagation. 
Let's leave it to **TRACER**.

---

@adriancole Let's agree to disagree.
Exactly _because_ this is a library, I want exceptions thrown from call() to propagate up, but also want close() exceptions to propagate (_if_ no call exception was thrown). That is expected external behaviour.
I'll gladly take some effort to provide it once instead of having people wonder why the close() exception got swallowed later.
There are unit tests now.